iavl/iavl_range_proofs.go

```diff
@@ -0,0 +1,419 @@
+package iavl
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/tendermint/go-wire/data"
+)
+
+// KeyInRangeProof is an interface which covers both first-in-range and last-in-range proofs.
+type KeyInRangeProof interface {
+	Verify(startKey, endKey, key, value, root []byte) error
+}
+
+// KeyFirstInRangeProof is a proof that a given key is the first in a given range.
+type KeyFirstInRangeProof struct {
+	KeyExistsProof `json:"key_proof"`
+
+	Left  *PathWithNode `json:"left"`
+	Right *PathWithNode `json:"right"`
+}
+
+// String returns a string representation of the proof.
+func (proof *KeyFirstInRangeProof) String() string {
+	return fmt.Sprintf("%#v", proof)
+}
+
+// Verify that the first in range proof is valid.
+func (proof *KeyFirstInRangeProof) Verify(startKey, endKey, key, value []byte, root []byte) (err error) {
```

Comment:
Minor nit in this signature, could we make the return just `error` since for all the cases we either make the error or do a naked return. If we change it to just `error` so `func (proof *KeyFirstInRangeProof) Verify(startKey, endKey, key, value []byte, root []byte) error {`
then for all the naked returns in this function, make them `return nil`